### ISSUE TITLE
Improve animations frame duration (animSpeed) handling 

### ISSUE THREAD

**Issue by warang580** [replit/kaboom#35]
I've been trying to use animation in a game and I've noticed that you can only use one animSpeed for all the frames/animations of a sprite.

For example, here I have made a heart collectible and 3 animations with different frame durations that I would want to use inside Kaboom as-is. "Spawn" frames are 200ms long, "idle" are 800ms long, and "pick" are 100ms long. 

![image](https://user-images.githubusercontent.com/233165/116010638-81b1a780-a620-11eb-8232-cfbed34f7e19.png)

But I've noticed that you can only set a animSpeed for the whole sprite and only when you're adding a sprite to the scene.

I think it could be nice if you could set animation frame durations at 3 points in time :

- when loading a sprite, with loadSprite() and loadAseprite()
- when initialising a sprite, with sprite()
- when playing an animation, with play(), so you can speed up an animation based on some conditions

As an API, I thought about using something like :

Let say you have a sprite with two animations A and B and 5 frames like A1, A2, A3, B4, B5
You could say `{durations: {A: 200, B: 300}}` or `{durations: {A: 200, B: [125, 175]}}` or `{durations: [200, 200, 200, 300, 300]}` (like an aseprite files contains) so you can handle all level of customisations.

`play(animation, loop = true)` for sprites would transform into `play(animation, options = {})`
with options being something `{loop: <boolean>, durations: <durationData>, restart: <boolean>}`

I'm introducing restart because I find it strange to handle this myself:
```js
action("player", () => {
  if (! keyIsDown("left") && ! keyIsDown("right")) {
    // this might be kaboom's work
    if (player.curAnim() != 'idle') {
      player.play("idle", true)
    }
}
```

It would be nice that by default it does not switch if animation is already playing, unless you explicitely say to restart the animation.

If these changes interest you, I could make a PR.

**Comment by slmjkdbtl** [comment]
Sorry missed this one. These just implemented and published to `2000.0.0-beta.15`. Now you can define properties like `loop` and `speed` in both load step and play step:

```js
// speed is in frames per second
loadSprite("hero", "hero.png", {
	sliceX: 9,
	anims: {
		idle: { from: 0, to: 3, speed: 3, loop: true },
		run: { from: 4, to: 7, speed: 10, loop: true },
		hit: 8,
	},
});

// to use one off custom properties
player.play("run", { speed: 20, loop: false });
```

**Comment by warang580** [comment]
Amazing, keep up the great work !
